python/ray/data/_internal/sort.py

```diff
@@ -209,7 +215,12 @@ def sort_impl(
     # Use same number of output partitions.
     num_reducers = num_mappers
     # TODO(swang): sample_boundaries could be fused with a previous stage.
-    boundaries = sample_boundaries(blocks_list, sort_key, num_reducers, ctx)
+    if not sort_key.boundaries:
+        boundaries = sample_boundaries(blocks_list, sort_key, num_reducers, ctx)
+    else:
+        boundaries = [(b, ) for b in sort_key.boundaries]
```

Comment:
Shouldn't each tuple have a start and end boundary?

Reply:
Thank you for your attention. Each tuple in the boundaries here only needs to provide start. This form is consistent with the form of boundaries returned by the sample_boundaries function in line 219.